fuel/src/main/kotlin/com/github/kittinunf/fuel/util/Base64.kt

```diff
@@ -0,0 +1,161 @@
+package com.github.kittinunf.fuel.util
+
+/**
+ * Inspired From https://github.com/square/okio/blob/master/okio/src/main/kotlin/okio/-Base64.kt
+ */
+
+import java.lang.System.arraycopy
+
+fun ByteArray.encodeBase64(): ByteArray = encodeBase64ToArray()
+fun ByteArray.encodeBase64Url(): ByteArray = encodeBase64ToArray(map = BASE64_URL_SAFE)
+fun String.encodeBase64ToString(): String = String(toByteArray().encodeBase64())
+fun String.encodeBase64UrlToString(): String = String(toByteArray().encodeBase64Url())
+
+fun String.decodeBase64(): ByteArray? = decodeBase64ToArray()?.let { it }
+fun String.decodeBase64ToString(): String? = decodeBase64ToArray()?.let { String(it) }
+
+private val BASE64 = byteArrayOf(
```

Comment:
I think we could improve this a bit because we keep repeating `toByte()`.

How about we use `CharProgression` to help us out?

``` Kotlin
object Base64 {
  private val regular = listOf(('A'..'Z'), ('a'..'z'), ('0'..'9'). listOf('+', '/'))
  private val urlSafe = ....

  val BASE64 = regular.flatten().map { it.toByte() }  
  val URL_BASE64 = urlSafe.flatten().map { it.toByte() }
}
```

Reply:
I See what you meant. I have to look at it later on the day.